### PR TITLE
Event - onHear

### PR DESCRIPTION
**Pull request by EvilHero90**
Creatures (Player/Monster/NPC) can now be responsive through other Creatures talking.
There's to much possibilities to even list up on what could be achieved with this.

### CODE REVIEW

File: src/events.cpp [hunk 1]
```diff
@@ -245,6 +247,37 @@ ReturnValue Events::eventCreatureOnTargetCombat(Creature* creature, Creature* ta
 	return returnValue;
 }
 
+void Events::eventCreatureOnHear(Creature* creature, Creature* speaker, const std::string words, SpeakClasses type, Position pos)
```
Comment (TheSumm) [inline src/events.cpp]: `words` should be a reference.

File: src/events.cpp [hunk 2]
```diff
@@ -245,6 +247,37 @@ ReturnValue Events::eventCreatureOnTargetCombat(Creature* creature, Creature* ta
...
+
+	LuaScriptInterface::pushUserdata<Creature>(L, creature);
+	LuaScriptInterface::setCreatureMetatable(L, -1, creature);
+
+	LuaScriptInterface::pushUserdata<Creature>(L, speaker);
+	LuaScriptInterface::setCreatureMetatable(L, -1, speaker);
+
+	LuaScriptInterface::pushString(L, words);
+	lua_pushnumber(L, type);
+	LuaScriptInterface::pushPosition(L, pos);
```
Comment (TheSumm) [inline src/events.cpp]: I don't see the purpose of the `pos` parameter. You can easily retrieve that information in the event by calling `speaker:getPosition()`.

File: src/events.h [hunk 3]
```diff
@@ -72,6 +74,7 @@ class Events
 		bool eventCreatureOnChangeOutfit(Creature* creature, const Outfit_t& outfit);
 		ReturnValue eventCreatureOnAreaCombat(Creature* creature, Tile* tile, bool aggressive);
 		ReturnValue eventCreatureOnTargetCombat(Creature* creature, Creature* target);
+		void eventCreatureOnHear(Creature* creature, Creature* speaker, const std::string words, SpeakClasses type, Position pos);
```
Comment (nekiro) [inline src/events.h]: if you arent removing that position, it should be passed by reference and const like words (which isnt)

**Comment by soul4soul** [comment]
Cool idea to add onHear event. I can imagine a quest/task where a creature needs to be lured to a NPC to scare them etc...

File: src/events.cpp (hunk 2 above)
Comment (EvilHero90) [reply src/events.cpp]: I know that you can do it that way aswell but sometimes it's not bad to have the extra param

File: src/events.h (hunk 3 above)
Comment (EvilHero90) [reply src/events.h]: done

File: src/events.cpp (hunk 1 above)
Comment (EvilHero90) [reply src/events.cpp]: done

File: src/events.cpp (hunk 2 above)
Comment (WibbenZ) [reply src/events.cpp]: I agree, there is no need for it.
All you are doing is sending speaker:getPosition() but from C++ as creature->getPosition()
If the function is needed for a code we can do it via Lua.

Comment (Leo32onGIT) [reply src/events.cpp]: I'd have to agree with @TheSumm here - the position stuff is unnecessary.
The concept of this is fantastic though.

**Pull request merged**